internal/controllers/device_data_controller.go

```diff
@@ -318,11 +326,55 @@ func (udc *UserDevicesController) QueryDeviceErrorCodes(c *fiber.Ctx) error {
 		return err
 	}
 
+	q := &models.ErrorCodeQuery{ID: ksuid.New().String(), UserDeviceID: udi, ErrorCodes: req.ErrorCodes, QueryResponse: chtResp}
+	err = q.Insert(c.Context(), udc.DBS().Writer, boil.Infer())
+
+	if err != nil {
+		// TODO - should we return an error for this or just log it
+		udc.log.Err(err).Msg("Could not save user query response")
+	}
+
 	return c.JSON(&QueryDeviceErrorCodesResponse{
 		Message: chtResp,
 	})
 }
 
+// GetUserDevicesErrorCodeQueries godoc
+// @Description Returns all error codes queries for user devices
+// @Tags        user-devices
+// @Success     200 {object} controllers.GetUserDevicesErrorCodeQueriesResponse
+// @Security    BearerAuth
+// @Router      /user/devices/error-codes [get]
+func (udc *UserDevicesController) GetUserDevicesErrorCodeQueries(c *fiber.Ctx) error {
+	userID := helpers.GetUserID(c)
+
+	userDevices, err := models.UserDevices(
+		models.UserDeviceWhere.UserID.EQ(userID),
+		qm.Load(models.UserDeviceRels.ErrorCodeQueries),
```

Comment:
I think we need to put the order by inside of here. I think we're ordering the _devices_ here but not the queries.